dataset/src/main/scala/frameless/functions/Lit.scala

```diff
@@ -1,30 +1,35 @@
 package frameless.functions
 
-import frameless.TypedEncoder
 import org.apache.spark.sql.catalyst.InternalRow
 import org.apache.spark.sql.catalyst.expressions.codegen._
-import org.apache.spark.sql.catalyst.expressions.{Expression, Literal, NonSQLExpression}
+import org.apache.spark.sql.catalyst.expressions.{Expression, NonSQLExpression}
 import org.apache.spark.sql.types.DataType
 
-case class FramelessLit[A](obj: A, encoder: TypedEncoder[A]) extends Expression with NonSQLExpression {
-  override def nullable: Boolean = encoder.nullable
-  override def toString: String = s"FramelessLit($obj)"
+case class Lit[T <: AnyVal] private[frameless] (
+    val dataType: DataType,
+    val nullable: Boolean,
+    toCatalyst: CodegenContext => ExprCode,
+    show: () => String)
+    extends Expression
+    with NonSQLExpression {
+  override def toString: String = s"FramelessLit(${show()})"
 
+  @SuppressWarnings(Array("AsInstanceOf", "MethodReturningAny"))
```

Comment:
Not required for this PR I guess. But can be a part of https://github.com/typelevel/frameless/pull/557 in case we decide to merge it.

Reply:
Btw what about these tiny comments? They were all marked as resolved but it looks like no changes were pushed / no answeres posted.